asyncapi/github/listener.bal

```diff
@@ -15,25 +16,41 @@
 // under the License.
 
 import ballerina/cloud;
-import ballerina/http;
+import ballerina/log;
+import ballerina/websub;
+
+const string GITHUB_REST_API_BASE_URL = "https://api.github.com";
+public const string HUB = "https://api.github.com/hub";
 
 @display {label: "GitHub", iconPath: "docs/icon.png"}
 public class Listener {
-    private http:Listener httpListener;
     private DispatcherService dispatcherService;
+    private websub:Listener websubListener;
+    private string? accessToken = ();
+    private websub:SubscriberServiceConfiguration config = {};
 
-    public function init(ListenerConfig listenerConfig = { webhookSecret: DEFAULT_SECRET }, @cloud:Expose int|http:Listener listenOn = 8090) returns error? {
-        if listenOn is http:Listener {
-            self.httpListener = listenOn;
-        } else {
-            self.httpListener = check new (listenOn);
-        }
-        self.dispatcherService = new DispatcherService(listenerConfig);
+    public function init(@cloud:Expose int listenOn = 8090) returns error? {
+        self.websubListener = check new (listenOn);
+        self.dispatcherService = new DispatcherService();
     }
 
     public isolated function attach(GenericServiceType serviceRef, () attachPoint) returns @tainted error? {
         string serviceTypeStr = self.getServiceTypeStr(serviceRef);
         check self.dispatcherService.addServiceRef(serviceTypeStr, serviceRef);
+
+        websub:SubscriberServiceConfiguration? configuration = retrieveSubscriberServiceAnnotations(serviceRef);
+        if (configuration is websub:SubscriberServiceConfiguration) {
+            self.config = <websub:SubscriberServiceConfiguration>configuration;
+            if (self.config?.httpConfig is http:ClientConfiguration) {
+                http:ClientConfiguration clientConfig = <http:ClientConfiguration>self.config?.httpConfig;
```

Comment:
Let's try to use ensureType() instead of manual casting wherever possible.

Reply:
Updated in https://github.com/ballerina-platform/asyncapi-triggers/commit/e0c56638e7d25694ff986c19c894b01fc776abb8